metals/src/main/scala/scala/meta/internal/metals/MtagsResolver.scala

```diff
@@ -0,0 +1,64 @@
+package scala.meta.internal.metals
+
+import java.util.concurrent.ConcurrentHashMap
+
+import scala.util.control.NonFatal
+
+import coursierapi.error.SimpleResolutionError
+
+class MtagsResolver {
+
+  private val states =
+    new ConcurrentHashMap[String, Either[Long, MtagsBinaries]]()
+
+  def resolve(scalaVersion: String): Option[MtagsBinaries] = {
+    def fetch(): Either[Long, MtagsBinaries] = {
+      try {
+        val jars = Embedded.downloadMtags(scalaVersion)
```

Comment:
Maybe we could do a retry also just in case?

Reply:
Good point. Will change it a bit to retry it in case if was a networking exception or smth like that.